sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java

```diff
@@ -126,9 +126,15 @@ public static void startCassandra() throws Exception {
 
   @AfterClass
   public static void stopCassandra() {
-    EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
-    session.close();
-    cluster.close();
+    if (cluster != null && session != null) {
+      EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
+      session.close();
+      cluster.close();
+    } else {
+      if (cluster != null){
```

Comment:
this validation is unneeded because is covered by the previous if. Probably is simpler to let it only with if (cluster != null) { ... } and do if (session != null) { ,,, } before.

Reply:
No, it is not the same validation. the  `if (cluster != null)` is under an `else`. So it is the code for the case `session == null  && cluster != null` i.e. the cluster is created but not the session. This is the case @mxn reported.
Your suggestion is not correct either, because both cluster and session need to be not null to clean cassandra, and we cannot close cluster and session beforewards because they are needed for the cleaning process.